ha/core/cluster/cluster_manager.py

```diff
@@ -180,9 +188,86 @@ def add_node(self, node):
         Log.info(f"Cluster status output after add node: {_output}, {_err}, {_rc}")
 
+    def get_nodes_status(self):
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_STATUS_NODES, check_error=False)
+
+        self.active_nodes = self.standby_nodes = self.offline_nodes = "false"
+        for status in _output.split("\n"):
+            nodes = status.split()
+            if len(nodes) > 0:
+                if nodes[0] == "Online:" and len(nodes) > 1:
+                    self.active_nodes = "true"
+
+                if nodes[0] == "Standby" and nodes[1] == "with" and nodes[2] == "resource(s)" and nodes[3] == "running:" and len(nodes) > 4:
+                    self.active_nodes = "true"
+
+                if((nodes[0] == "Maintenance:") and (len(nodes) > 1)):
+                    self.active_nodes = "true"
+
+                if((nodes[0] == "Standby:") and (len(nodes) > 1)):
+                    self.standby_nodes = "true"
+
+                if((nodes[0] == "Offline:") and (len(nodes) > 1)):
+                    self.offline_nodes = "true"
+
+
     def start(self):
-        # TODO Add wrapper to hctl pcswrap
-        raise HAUnimplemented("This feature is not supported...")
+
+        Log.debug("Executing cortxha cluster start")
+
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_CLUSTER_STATUS, check_error=False)
+        if _rc != 0:
+            if(_err.find("No such file or directory: 'pcs'") != -1):
+                Log.error("Cluster failed to start; pcs not installed ")
+                raise Exception("Cluster failed to start; pcs not installed")
+            # if cluster is not running; start cluster
+            elif(_err.find("cluster is not currently running on this node") != -1):
+                self._execute.run_cmd(const.PCS_CLUSTER_START, check_error=False)
+                Log.info("cluster started ; waiting for nodes to come online ")
+                # It takes nodes 30 seconds to come to their original state after cluster is started
+                # observation on a 2 node cluster
+                # wait for upto 100 sec for nodes to come to active states (online / maintenance mode)
+                time.sleep(10)
+                self.get_nodes_status()
+                retries = 18
+                while  self.active_nodes == "false" and retries > 0:
+                    time.sleep(5)
+                    self.get_nodes_status()
+                    retries -= 1
+
+        else:
+            #If cluster is running, but all nodes are either Offline or in Standby mode;
+            #start the nodes
+            self.get_nodes_status()
+            if self.active_nodes == "false":
+                if self.standby_nodes == "true":
+                    # issue pcs cluster unstandby
+                    _output, _err, _rc = self._execute.run_cmd(const.PCS_CLUSTER_UNSTANDBY, check_error=False)
+
+                if self.offline_nodes == "true":
```

Comment:
If there are 3 nodes, two is in standby and third is offline. This code will call "cluster unstandby" and then cluster start. I think there is a problem. I guess, we should not take any action against offline nodes.

Reply:
@ajaysrivas I did this to have a better chance of having some active nodes in the system. 
If we are not going to take any action for offline nodes, I will remove the code under if self.offline_nodes == "true": altogether.

Hope that is fine

Reply:
@ajaysrivas , in the 3 node scenario  mentioned above:
   if third node (node3) was online and at that time cluster was stopped on third node.
   After this if we issue 'pcs node status' on node1 or node2, it will show the node3 status as Offline.
starting cluster (pcs cluster start --all) will make this node Online

However as per our discussion on this in the meeting, this particular scenario can be ignored. 

So removing the offline_nodes  case altogether.